sgkit/io/dataset.py

```diff
@@ -18,17 +24,25 @@ def save_dataset(ds: Dataset, path: PathType, **kwargs: Any) -> None:
         Dataset to save.
     path
         Path to directory in file system to save to.
+    storage_options:
+        Any additional parameters for the storage backend (see ``fsspec.open``).
     kwargs
         Additional arguments to pass to :meth:`xarray.Dataset.to_zarr`.
     """
-    store = str(path)
+    if isinstance(path, str):
```

Comment:
My recollection was that the mapper should be created here to do any context management (I thought something couldn't be left open without throwing a warning?).  I don't see that anywhere in this PR though so is it still true that `path` can't be `store: Union[MutableMapping, Path, str]` like in `to_zarr`?

Reply:
> My recollection was that the mapper should be created here to do any context management (I thought something couldn't be left open without throwing a warning?).

The mapper doesn't have a close method, so it's not used within a context manager itself. Instead, each get and set item call on the mapper uses a context manager internally to ensure resources are closed. There is a bug in the latest fsspec release (fixed in intake/filesystem_spec#463) where a context manager is *not* used for the get item call, which is why there are lots of warnings when running tests.

Re `MutableMapping` - we should probably allow that too.

Reply:
> Re MutableMapping - we should probably allow that too.

👍

Reply:
Added in https://github.com/pystatgen/sgkit/pull/423/commits/55fdd4e54e5f3d6d4eb31462126c254d77928b29